### PR TITLE
Warn for unexpected argument combinations

### PR DESCRIPTION
**Pull request by shirayu**
# What does this PR do?

Warn for unexpected argument combinations:

- `padding is True` and ``max_length is not None``
- `padding is True` and ``pad_to_max_length is not False``

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
    - https://discuss.huggingface.co/t/migration-guide-from-v2-x-to-v3-x-for-the-tokenizer-api/55
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by shirayu** [comment]
@sgugger 
Could you review this PR?
Thank you.

**Review by sgugger — APPROVED** [review]
Looks good on my side! Thanks a lot for your PR!

File: src/transformers/tokenization_utils_base.py [hunk 1]
```diff
@@ -2222,6 +2222,11 @@ def _get_padding_truncation_strategies(
                 padding_strategy = PaddingStrategy.MAX_LENGTH
         elif padding is not False:
             if padding is True:
+                if verbose:
+                    if max_length is not None:
+                        warnings.warn("`max_length` is ignored when `padding`=`True`.")
+                    if old_pad_to_max_length is not False:
+                        warnings.warn("`pad_to_max_length` is ignored when `padding`=`True`.")
```
Comment (LysandreJik) [inline src/transformers/tokenization_utils_base.py:2229]: This argument is `True` when the following kwarg was passed as `True`: `pad_to_max_length`. Could you edit the warning to show that?

**Review by LysandreJik — COMMENTED** [review]
Looks great, just a small modification of the second warning :)

File: src/transformers/tokenization_utils_base.py (hunk 1 above)
Comment (shirayu) [reply src/transformers/tokenization_utils_base.py]: I updated the line.
Is it what you intended?

Comment (LysandreJik) [reply src/transformers/tokenization_utils_base.py]: Thank you!

**Pull request merged**